python/cuml/manifold/umap.pyx

```diff
@@ -313,6 +316,8 @@ class UMAP(Base):
         self.umap_params = <size_t> umap_params
 
         self.callback = callback  # prevent callback destruction
+        self.X_m = None
```

Comment:
I'm still seeing this in the CI log so I think UMAP is still having pickling issues:
```
self = <class 'cudf.core.dataframe.DataFrame'>, data = None, index = None

columns = None, nan_as_null = False



    @classmethod

    def from_gpu_matrix(

        self, data, index=None, columns=None, nan_as_null=False

    ):

        """Convert from a numba gpu ndarray.

    

        Parameters

        ----------

        data : numba gpu ndarray

        index : str

            The name of the index column in *data*.

            If None, the default index is used.

        columns : list of str

            List of column names to include.

    

        Returns

        -------

        DataFrame

        """

>       if data.ndim != 2:

E       AttributeError: 'NoneType' object has no attribute 'ndim'
```

This is happening because I'm using `hasattr` in the serialization functions and you are setting them to None in the constructor, causing it to call `None.from_gpu_matrix()` in the deserialization function. 

I think we should either 1) not set the values to None in the constructor, or 2) check for `hasattr() and is not None`. Otherwise, I think this PR is ready to go.